src/coreclr/jit/compiler.cpp

```diff
@@ -5235,12 +5231,26 @@ void Compiler::compCompile(void** methodCodePtr, uint32_t* methodCodeSize, JitFl
     rat.Run();
 
 #if defined(TARGET_WASM)
-    lvaMarkLocalVars();
+    if (opts.OptimizationEnabled())
+    {
+        lvaMarkLocalVars(); // For optimization only
+    }
+
+    Llvm* llvm = new Llvm(this);
+    auto placeAndConvertShadowStackLocalsPhase = [llvm]() {
+        llvm->PlaceAndConvertShadowStackLocals();
+    };
+    DoPhase(this, PHASE_SHDWSTK_SETUP, placeAndConvertShadowStackLocalsPhase);
 
+    lvaMarkLocalVars();  // For SSA i. e. correctness
```

Comment:
```suggestion
    lvaMarkLocalVars();  // For SSA.
```
Don't think there is a need to elaborate, the lack of `if (OptimizationEnabled())` implies that already.